src/FakeItEasy/ReturnValueConfigurationExtensions.cs

```diff
@@ -308,8 +308,12 @@ public static void ReturnsNextFromSequence<T>(this IReturnValueConfiguration<T>
         {
             Guard.AgainstNull(configuration, nameof(configuration));
 
-            var queue = new Queue<T>(values);
-            configuration.ReturnsLazily(x => queue.Dequeue()).NumberOfTimes(queue.Count);
+            var queue = new ConcurrentQueue<T>(values);
+            configuration.ReturnsLazily(x =>
+            {
+                queue.TryDequeue(out T returnValue);
+                return returnValue;
+            }).NumberOfTimes(queue.Count);

```

Comment:
I was unable to get it to fail, but I could add some tests for this, if you like.
Sadly, they'd be completely nondeterministic.

Reply:
I'm not sure what the problem is here... How could this fail anyway? The queue isn't reference anywhere else, so there's no chance of another thread adding or removing values. Or am I missing something?

Reply:
`Queue.Dequeue` mutates the queue. If multiple threads call the configured method concurrently, the could (in theory) interfere with one another, no?

Reply:
Ah, yes. I was trying to find a scenario where the queue would be mutated by another thread, but didn't think of the simplest case. Never mind!

Reply:
Your mind is too advanced!